mentoring-app/src/app/courses-page/course/course.component.html

```diff
@@ -1,6 +1,24 @@
+<div class="section">
+    <div class="search">
+        <form class="seatch_field">
+                <mat-form-field class="search_input">
+                  <input matInput placeholder="Search" name="search" [(ngModel)]="searchInput">
+                </mat-form-field>
+
+                <button mat-raised-button class="search_button" (click)="onFindClick()">Find</button>
```

Comment:
Your onFindClick method doesn't filter courses indeed, because you search in coursesCopy array, but iterate over courses array.

Reply:
fixed